### PR TITLE
Prepared framework to run unittests

### PR DESCRIPTION
**Pull request by HaruChebrolu**
Signed-off-by: Chebrolu Harika <hchebrol@redhat.com>

# Description

Configured a baseline to run unit tests using pytest, unittest unit testing frameworks.

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: ceph/ceph_admin/add.py [hunk 1]
```diff
@@ -34,10 +34,11 @@ def add(self: DaemonProtocol, config: Dict):
...
 
-        pos_args = config["pos_args"]
-        node = pos_args[0]
-        host_id = get_node_by_id(self.cluster, node)
-        host = host_id.shortname
+        if config.get("pos_args"):
+            pos_args = config["pos_args"]
+            node = pos_args[0]
+            host_id = get_node_by_id(self.cluster, node)
+            host = host_id.shortname
```
Comment (psathyan) [inline ceph/ceph_admin/add.py:41]: This changes the behavior. We need it to raise an issue when `pos_args` is not provided. With the new change, no `AttributeError` would be raised.

**Review by psathyan — COMMENTED** [review]
This is an excellent initiative. Thanks for doing it.

One minor change request.

File: ceph/ceph_admin/add.py (hunk 1 above)
Comment (HaruChebrolu) [reply ceph/ceph_admin/add.py]: I have actually fixed this in second commit, but when I did rebase somehow hose changes were disappeared.
Anyways, I provided a workaround for that.

**Review by psathyan — COMMENTED** [review]
Thanks for putting this together.

Is it possible to remove `.coverage` file?

At present, we are using `pytest` for some of our unit tests. Here, we are using `unittest`... let me know your thoughts on if we need to migrate from `pytest`

**Comment by HaruChebrolu** [comment]
> Thanks for putting this together.
> 
> Is it possible to remove `.coverage` file?
> 
> At present, we are using `pytest` for some of our unit tests. Here, we are using `unittest`... let me know your thoughts on if we need to migrate from `pytest`

Removed .coverage. 
I used `unittest` because it is very easy to understand and read even for who are novice in writing unit tests. However, if we agree upon using `pytest`, I will refactor the code to `pytest`.

**Comment by psathyan** [comment]
> > Thanks for putting this together.
> > Is it possible to remove `.coverage` file?
> > At present, we are using `pytest` for some of our unit tests. Here, we are using `unittest`... let me know your thoughts on if we need to migrate from `pytest`
> 
> Removed .coverage. I used `unittest` because it is very easy to understand and read even for who are novice in writing unit tests. However, if we agree upon using `pytest`, I will refactor the code to `pytest`.

Let's use `pytest` as we have been using it for our unit tests. We can always fallback to `unittest` if we believe, we cannot scale.

**Comment by HaruChebrolu** [comment]
> > > Thanks for putting this together.
> > > Is it possible to remove `.coverage` file?
> > > At present, we are using `pytest` for some of our unit tests. Here, we are using `unittest`... let me know your thoughts on if we need to migrate from `pytest`
> > 
> > 
> > Removed .coverage. I used `unittest` because it is very easy to understand and read even for who are novice in writing unit tests. However, if we agree upon using `pytest`, I will refactor the code to `pytest`.
> 
> Let's use `pytest` as we have been using it for our unit tests. We can always fallback to `unittest` if we believe, we cannot scale.

Done.

File: requirements.txt [hunk 2]
```diff
@@ -5,5 +5,8 @@ isort
 flake8
 tox
 python-openstackclient
+pyOpenSSL
```
Comment (psathyan) [inline requirements.txt:8]: I believe this is already in `setup.py`. I think the time has come to write a `test-requirements.txt` which captures the requirements for gating purpose. We can pick this in another PR though. Just a thought.

**Review by psathyan — COMMENTED** [review]
Looks good to me

File: requirements.txt (hunk 2 above)
Comment (HaruChebrolu) [reply requirements.txt:8]: I will list down these in seperate test-requirements.txt in a seperate PR.

**Review by psathyan — APPROVED** [review]

**Pull request merged**